PerimeterXModule/Internals/PXClient/DefaultPxClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using Jil;
+using System.Threading;
+
+namespace PerimeterX
+{
+    public class DefaultPxClient : PxClient
+    {
+        private HttpClient httpClient;
+        private PXConfigurationWrapper pxConfig;
+
+        public DefaultPxClient(PXConfigurationWrapper pxConfig)
+        {
+            this.pxConfig = pxConfig;
+			var webRequestHandler = new WebRequestHandler
+			{
+				AllowPipelining = true,
+				UseDefaultCredentials = true,
+				UnsafeAuthenticatedConnectionSharing = true
+			};
+			httpClient = new HttpClient(webRequestHandler, true);
+
+			httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", pxConfig.ApiToken);
+			httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+			httpClient.DefaultRequestHeaders.ExpectContinue = false;
+        }
+
+        public PXDynamicConfiguration GetConfigurationRequest(string checksumParams)
+        {
+			string requestUrl = string.Format("{0}{1}{2}", PxConstants.REMOTE_CONFIGURATION_SERVER, PxConstants.REMOTE_CONFIGURATION_PATH, checksumParams);
+			Debug.WriteLine(string.Format("Get request for {0}", requestUrl), PxConstants.LOG_CATEGORY);
+
+			// Set request
+			var requestMessage = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+			CancellationTokenSource source = new CancellationTokenSource(TimeSpan.FromMilliseconds(pxConfig.ApiTimeout));
+			CancellationToken cancellationToken = source.Token;
+			var httpResponse = httpClient.SendAsync(requestMessage, cancellationToken).Result;
+
+            Debug.WriteLine(string.Format("response for {0} received status code {1}", requestUrl, httpResponse.StatusCode));
+
+			PXDynamicConfiguration pxDynamicConfiguration = null;
+            if (!httpResponse.StatusCode.Equals(HttpStatusCode.OK))
```

Comment:
the else if is not needed. if you check for !statuscode.ok then the else can only be statuscode.ok. In addition the return should be inside the if since the only other option is the else (which has return in it)